src/ray/raylet/lineage_cache.h

```diff
@@ -20,16 +20,14 @@ namespace raylet {
 enum class GcsStatus {
   /// The task is not in the lineage cache.
   NONE = 0,
-  /// The task is being executed or created on a remote node.
-  UNCOMMITTED_REMOTE,
-  /// The task is waiting to be executed or created locally.
-  UNCOMMITTED_WAITING,
-  /// The task has started execution, but the entry has not been written to the
-  /// GCS yet.
-  UNCOMMITTED_READY,
-  /// The task has been written to the GCS and we are waiting for an
-  /// acknowledgement of the commit.
+  /// The task is uncommitted. Unless there is a failure, we will expect a
```

Comment:
It's probably worth documenting the fact that we're making inequality comparisons between these states. Someone inserting a state probably won't think about the ordering.

Or we can just not use the comparisons and be more explicit.

Reply:
Thanks, I'll update the documentation.

I actually like having the comparisons in this case because it forces us to think about what state transitions are okay.